File: tasks/implementations/cpp/tasks/sentiment_analysis.cpp
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <unordered_set>

void run_sentiment_analysis(const std::string& size, const std::string& fixturesRoot) {
    static const std::unordered_set<std::string> positive = {"good", "great", "happy", "clean", "fast", "love"};
    static const std::unordered_set<std::string> negative = {"bad", "sad", "dirty", "slow", "hate", "poor"};
    std::ifstream file(fixturesRoot + "/generated/sentiment/" + size + ".txt");
    std::string line, token;
    long long total = 0;
    while (std::getline(file, line)) {
        std::stringstream ss(line);
        while (ss >> token) {
            if (positive.contains(token)) total += 1;
            else if (negative.contains(token)) total -= 1;
        }
    }
    std::cout << total << std::endl;
}

File: tasks/implementations/node/tasks/sentiment_analysis.js
const fs = require('fs');
const POSITIVE = new Set(['good', 'great', 'happy', 'clean', 'fast', 'love']);
const NEGATIVE = new Set(['bad', 'sad', 'dirty', 'slow', 'hate', 'poor']);
module.exports = function(size, fixturesRoot) {
  const lines = fs.readFileSync(`${fixturesRoot}/generated/sentiment/${size}.txt`, 'utf8').trim().split(/\r?\n/);
  let total = 0;
  for (const line of lines) {
    for (const token of line.split(/\s+/)) {
      if (POSITIVE.has(token)) total += 1;
      else if (NEGATIVE.has(token)) total -= 1;
    }
  }
  console.log(total);
};
